### PR TITLE
Port C FFI crate to maybenot-v2

### PR DESCRIPTION
**Pull request by hulthe**
Port the `maybenot-ffi` crate from v1 to v2, and expose the new events and actions.

This port simply uses the `std::time` types, but we may want to consider using something else. If it's possible to use a no_std time implementation, that would be preferable as it avoids linking rust std into the resulting static lib.

### CODE REVIEW

**Comment by hulthe** [comment]
As of right now, this compiles, but it hasn't been thoroughly tested. Therefore I'm leaving it as a draft PR until we get around to doing that.

**Comment by pylls** [comment]
Many thanks @hulthe ! Let me know when you think it's in a good shape.

**Comment by pylls** [comment]
Hi again, @hulthe. This is just a request for when you finish this PR: please also update the root repository README.md to include a link to this crate (like it was in v1). In getting v2.0 out there, I temporarily removed it since the v2 (now new main) branch started before v1 of the FFI crate was there.

File: crates/maybenot-ffi/README.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+## Building
+You need to have [rust](https://rustup.rs/) installed.
+`cbindgen` is also required: `cargo install --force cbindgen`
+Then just run `make` to build a static library at `maybenot-ffi/libmaybenot.a`.
+
+Arguments to `make`:
+- `CARGO` override the `cargo` command
+- `TARGET` override target architecture; cross-compile.
+- `PROFILE` override the cargo profile, valid options are `release` and `debug`.
```
Comment (faern) [inline crates/maybenot-ffi/README.md:15]: Could be useful to point out what the default is.

Comment (hulthe) [reply crates/maybenot-ffi/README.md]: Done

**Comment by faern** [comment]
I went ahead and merged this, since we are happy with this FFI. @pylls are you able to publish this update to `maybenot-ffi` to crates.io?

**Comment by pylls** [comment]
Excellent, thank you all! Sorted.

**Pull request merged**